storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -6788,13 +6790,16 @@ bool xb_init() {
   const char *mixed_options[4] = {NULL, NULL, NULL, NULL};
   int n_mixed_options;
 
+  if (opt_no_lock || opt_safe_slave_backup) opt_lock_ddl = false;
```

Comment:
What about master-master replication or topologies where there are writes on the slave?
Not sure we should implicitly disable lock_ddl with safe_slave_backup is used.

Reply:
> What about master-master replication or topologies where there are writes on the slave?
> Not sure we should implicitly disable lock_ddl with safe_slave_backup is used.
safe_slave_backup option should not be used for master-master replication because PXB designed to wait until Slave_open_temp_tables is zero which tells that it is for master-slave topology.

Reply:
So, here we are explicitly disabling `--lock-ddl` if we are in a replica. 
However, I can see backups failing because of that. One example is reporting a replica that is also used to take backup, sometimes to run a report one needs to aggregate data into a summary table that gets created and deleted after the reporting is done.
If one wants only `--safe-slave-backup` and no `--lock-ddl` it should run the backup with `--lock-ddl=FALSE --safe-slave-backup` . I don't know where this requirement of disabling `lock-ddl` came from, but by automatically disabling --lock-ddl if --safe-slave-backup is used, we are making the default less safe, which is the oposite of what this PR was meant for.

Reply:
I disabled lock-ddl with safe-slave-backup since safe-slave-back disable all DDL and DML by stopping Slave SQL thread.  So I thought, there won't be any SQL running. I didn't consider other scenarios. 
I agree to your point it makes more sense to enable lock-ddl by default and let the user disable if he wants.
Will fix code and push changes soon.

Reply:
STOP SLAVE SQL thread is now done before taking LIFB/LTFB.